src/caffe/data_transformer.cpp

```diff
@@ -165,8 +165,8 @@ void DataTransformer<Dtype>::Transform(const vector<Datum> & datum_vector,
   const int width = transformed_blob->width();
 
   CHECK_GT(datum_num, 0) << "There is no datum to add";
-  CHECK_LE(datum_num, num) <<
-    "The size of datum_vector must be smaller than transformed_blob->num()";
+  CHECK_EQ(datum_num, num) <<
```

Comment:
The reason behind that was that one can convert a Vector into a bigger Blob, but not the opposite. However it is ok either way.